arcgishub/hub.py

```diff
@@ -1,182 +1,133 @@
 from arcgis.gis import GIS
-from arcgis.features import FeatureLayer
-import json
+from arcgis._impl.common._mixins import PropertyMap
+import collections
 
-class Hub(object):
-    "Acceessing an individual hub and its events, indicators and initiatives"
+def _lazy_property(fn):
+    '''Decorator that makes a property lazy-evaluated.
+    '''
+    # http://stevenloria.com/lazy-evaluated-properties-in-python/
+    attr_name = '_lazy_' + fn.__name__
+
+    @property
+    def _lazy_property(self):
+        if not hasattr(self, attr_name):
+            setattr(self, attr_name, fn(self))
+        return getattr(self, attr_name)
+    return _lazy_property
 
+class Hub(object):
+    """Entry point. Acceessing an individual hub and its components"""
+    
     def __init__(self, url, username=None, password=None):
         self.url = url
         self._username = username
         self._password = password
-        self._org = GIS(self.url, self._username, self._password)
-    
+        self._initiative = None
+        self.org = GIS(self.url, self._username, self._password)
+        
+    def _org_id(self):
+        '''Return the Organization Id for this hub'''
```

Comment:
I think this is the `Enterprise Organization ID` for this Hub?

Reply:
Works for community organization as well.

Reply:
Ok - I see that `self._org` is whatever you used to create the Hub. However, this could be confusing if someone uses the community Org but then searchers for events or initiatives and there aren't any because they are only in the enterprise Org.

We should probably be explicit in storing both the `_enterpriseOrg` and `_communityOrg`. Then in searches we would be sure to search the `_enterpriseOrg` for now, and in the future both orgs since there could be initiatives in each, but there will (likely) only ever be events in the Enterprise org.

Reply:
to be explicit, the following lines should all result in the same "Hub":

```
myHub1 = hub.Hub("https://dc.maps.arcgis.com", username, password)
myHub2 = hub.Hub("https://community.maps.arcgis.com", username, password)
myHub3 = hub.Hub("https://www.arcgis.com", username, password)
myHub4 = hub.Hub("https://opendata.dc.gov", username, password)

myHub1.orgs.enterprise #=> https://dc.maps.arcgis.com
myHub1.orgs.community #=> https://community.maps.arcgis.com

myHub2.orgs.enterprise #=> https://dc.maps.arcgis.com
myHub2.orgs.community #=> https://community.maps.arcgis.com

myHub3.orgs.enterprise #=> https://dc.maps.arcgis.com
myHub3.orgs.community #=> https://community.maps.arcgis.com

myHub4.orgs.enterprise #=> https://dc.maps.arcgis.com
myHub4.orgs.community #=> https://community.maps.arcgis.com
```

Reply:
I see, yes I'll change that.

Reply:
We can address the latter in a follow up PR for #35